ci-operator/step-registry/aws/provision/dns-firewall/aws-provision-dns-firewall-commands.sh

```diff
@@ -0,0 +1,186 @@
+  "cart-rhcos-ci.s3.amazonaws.com",
+  "openshift.org",
+  "registry.access.redhat.com",
+  "registry.connect.redhat.com",
+  "pull.q1w2.quay.rhcloud.com",
+  "q1w2.quay.rhcloud.com",
+  "www.okd.io",
+  "www.redhat.com",
+  "aws.amazon.com",
+  "catalog.redhat.com",
+
+  "cert-api.access.redhat.com",
+  "api.access.redhat.com",
+  "infogw.api.openshift.com",
+  "console.redhat.com",
+  "cloud.redhat.com",
+  "observatorium-mst.api.openshift.com",
+  "observatorium.api.openshift.com",
+
+  "ec2.amazonaws.com",
+  "events.'${REGION}'.amazonaws.com",
+  "iam.amazonaws.com",
+  "route53.amazonaws.com",
+  "sts.amazonaws.com",
+  "sts.'${REGION}'.amazonaws.com",
+  "tagging.us-east-1.amazonaws.com",
+  "ec2.'${REGION}'.amazonaws.com",
+  "elasticloadbalancing.'${REGION}'.amazonaws.com",
+  "servicequotas.'${REGION}'.amazonaws.com",
+  "tagging.'${REGION}'.amazonaws.com",
+
+  "mirror.openshift.com",
+  "storage.googleapis.com",
+  "api.openshift.com",
+
+  "api.pagerduty.com",
+  "events.pagerduty.com",
+  "api.deadmanssnitch.com",
+  "nosnch.in",
+  "osdsecuritylogs.splunkcloud.com",
+  "http-inputs-osdsecuritylogs.splunkcloud.com",
+  "sftp.access.redhat.com"
+')
+
+DOMAIN_LIST=${OCP_DOMAIN_ALLOW_LIST}
+if [[ ${MANAGED_CLUSTER} == "yes" ]]; then
+  DOMAIN_LIST=${OSD_DOMAIN_ALLOW_LIST}
+fi
+
+function aws_add_param_to_json() {
+    local k="$1"
+    local v="$2"
+    local param_json="$3"
+    if [ ! -e "$param_json" ]; then
+        echo -n '[]' > "$param_json"
+    fi
+    cat <<< "$(jq  --arg k "$k" --arg v "$v" '. += [{"ParameterKey":$k, "ParameterValue":$v}]' "$param_json")" > "$param_json"
+}
+
+# Create DNS firewall
+UNIQUE_HASH=$(head /dev/urandom | tr -dc A-Za-z0-9 | head -c 4)
+STACK_NAME="${NAMESPACE}-${UNIQUE_HASH}-dns-firewall"
+dns_firewall_domain_list_name="${STACK_NAME}-domain-list"
+dns_firewall_rule_group_name="${STACK_NAME}-rule-group"
+RULE_GROUP_ALLOWED=$(echo -e '
+{
+    "AWSTemplateFormatVersion": "2010-09-09",
+    "Description": "",
+    "Resources": {
+        "ResolverFirewallDomainList": {
+            "Type": "AWS::Route53Resolver::FirewallDomainList",
+            "Properties": {
+                "Name": "'${dns_firewall_domain_list_name}'",
+                "Domains": ['${DOMAIN_LIST}']
+            }
+        },
+        "ResolverFirewallRuleGroup": {
+            "Type": "AWS::Route53Resolver::FirewallRuleGroup",
+            "Properties": {
+                "Name": "'${dns_firewall_rule_group_name}'",
+                "FirewallRules": [
+                    {
+                        "Priority": 1,
+                        "Action": "ALLOW",
+                        "FirewallDomainListId": {
+                            "Ref": "ResolverFirewallDomainList"
+                        }
+                    }
+                ]
+            }
+        }
+    },
+    "Parameters": {},
+    "Metadata": {},
+    "Conditions": {}
+}')
+
+echo "Create DNS firewall"
+aws --region "${REGION}" cloudformation create-stack \
```

Comment:
As the configuration is simple, I think the aws util is more proper/simple than CF template.
Or we can consider put all configurations (including following firewall rule group association) into CF template.

Reply:
@yasun1 I wrote a CF template for DNS Firewall setup, for your reference : https://gist.github.com/yunjiang29/7562d53114345c79f8d6c5a0e195c5c4

Reply:
Using the stack, it could be easily for us to delete the created resource.

Reply:
> I wrote a CF template for DNS Firewall setup, for your reference : https://gist.github.com/yunjiang29/7562d53114345c79f8d6c5a0e195c5c4

That is great. Let me check.